### PR TITLE
Timer

### PR DESCRIPTION
**Pull request by Aboorva1**
Added timer to quiz page
Added a column to add minutes to quiz
Quiz will get auto submitted and answers will be saved based on hidden field value
Made view changes in result page

### CODE REVIEW

File: app/controllers/quizzes_controller.rb [hunk 1]
```diff
@@ -5,9 +5,9 @@ class QuizzesController < ApplicationController
 
   def index
     if(params.has_key?(:category_name))
-      @quizzes = Quiz.where(category_name: params[:category_name]).paginate(page: params[:page], per_page: 4).order("created_at desc")
+      @quizzes = Quiz.where(category_name: params[:category_name]).paginate(page: params[:page], per_page: 5).order("created_at desc")
     else
-      @quizzes = Quiz.all.paginate(page: params[:page], per_page: 4).order("created_at desc")
+      @quizzes = Quiz.all.paginate(page: params[:page], per_page: 5).order("created_at desc")
```
Comment (iMagesh) [inline app/controllers/quizzes_controller.rb:10]: why does the alignment look off?

File: app/controllers/user_quizzes_controller.rb [hunk 2]
```diff
@@ -1,76 +1,62 @@
...
-      flash[:error] = "Please select one option"
-      redirect_back(fallback_location: root_path)
+      if (params[:answers].present?)
+        quiz_id = save_user_answer(params[:answers])
+        answers_count = (params[:page].to_i) + 1
+        save_user_quiz(answers_count)
+      else
+        answers_count = (params[:page].to_i)
+        save_user_quiz(answers_count)
+      end
```
Comment (iMagesh) [inline app/controllers/user_quizzes_controller.rb:33]: can you refactor this code? simplify it?

File: app/views/user_quizzes/show.html.erb [hunk 3]
```diff
@@ -1,3 +1,44 @@
...
+      if (--timer < 0) {
+        document.getElementById("auto_save").value = true;
+        document.getElementById("quiz_form").submit();
+      }
+    window.localStorage.setItem("seconds",seconds)
+    window.localStorage.setItem("minutes",minutes)
+    }, 1000);
+  }
+
+  var time_limit = "<%= @quiz.minutes %>";
```
Comment (iMagesh) [inline app/views/user_quizzes/show.html.erb:23]: can you move this script to a separate js file?

File: app/controllers/quizzes_controller.rb (hunk 1 above)
Comment (Aboorva1) [reply app/controllers/quizzes_controller.rb:10]: It is aligned properly in code

**Comment by iMagesh** [comment]
@Aboorva1 are you working on this? let me know once you have updated the changes so i can review and merge

**Pull request closed without merging**